pkg/sql/sqlstats/insights/integration/integration_test: TestFailedInsights failed

pkg/sql/sqlstats/insights/integration/integration_test.TestFailedInsights [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11746794?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11746794?buildTab=artifacts#/) on master @ [d98ce4fe44effcc24d26ba4b25acbb744f2fbabb](https://github.com/cockroachdb/cockroach/commits/d98ce4fe44effcc24d26ba4b25acbb744f2fbabb):


```
=== RUN   TestFailedInsights
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/f75d8af2c2b6af04e0405b19fd4d23f3/logTestFailedInsights3279048800
    test_log_scope.go:81: use -show-logs to present logs inline
    insights_test.go:335: condition failed to evaluate within 1s: expected problems to be '{FailedExecution}', but was '{SlowExecution,FailedExecution}'. stmts: BEGIN; SHOW DATABASES; SELECT (2/0);
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f75d8af2c2b6af04e0405b19fd4d23f3/logTestFailedInsights3279048800
--- FAIL: TestFailedInsights (8.03s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cluster-observability
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFailedInsights.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31451

--- a/pkg/sql/sqlstats/insights/integration/insights_test.go
+++ b/pkg/sql/sqlstats/insights/integration/insights_test.go
@@ -352,7 +352,12 @@ func TestFailedInsights(t *testing.T) {
 			}
 
 			if problems != tc.problems {
-				return fmt.Errorf("expected problems to be '%s', but was '%s'. stmts: %s", tc.problems, problems, tc.stmts)
+				// During tests some transactions can stay open for longer, adding an extra `SlowExecution` to the problems
+				// list. This checks for that possibility.
+				withSlow := strings.Replace(tc.problems, "{", "{SlowExecution,", -1)
+				if problems != withSlow {
+					return fmt.Errorf("expected problems to be '%s', but was '%s'. stmts: %s", tc.problems, problems, tc.stmts)
+				}
 			}
 
 			if status != tc.txnStatus {

